core_impl/src/main/java/com/google/instrumentation/trace/SpanImpl.java

```diff
@@ -303,20 +330,45 @@ public void end(EndSpanOptions options) {
     void onEnd(SpanImpl span);
   }
 
+  private static final class AttributesWithCapacity extends LinkedHashMap<String, AttributeValue> {
+    private int capacity;
+    private int totalRecordedAttributes = 0;
+    private static final long serialVersionUID = 42L;
```

Comment:
I think that extending a collection or other serializable class isn't very common, so people might expect that a class with a serialVersionUID is supposed to be serialized.  A comment saying that the class shouldn't be serialzed would also help explain why the class has a non-serializable element type.